homeassistant/components/netatmo/climate.py

```diff
@@ -508,3 +519,46 @@ async def _async_service_set_temperature_with_time_period(
     async def _async_service_clear_temperature_setting(self, **kwargs: Any) -> None:
         _LOGGER.debug("Clearing %s temperature setting", self.device.entity_id)
         await self.device.async_therm_home()
+
+    async def _async_service_set_scheduled_room_temperature(
+        self, **kwargs: Any
+    ) -> None:
+        temperature = kwargs[ATTR_TEMPERATURE]
+        zone_name = kwargs[ATTR_ZONE_NAME]
+
+        schedule = self._get_active_schedule()
+        if not schedule:
+            _LOGGER.error("Could not determine active schedule")
+            return
+
+        selected_zone = None
+        for zone in schedule.zones:
+            if zone.name == zone_name:
+                selected_zone = zone
+                break
+
+        if not selected_zone:
+            _LOGGER.error("%s is not a valid zone", zone_name)
+            return
+
+        await self.device.home.async_set_schedule_temperatures(
+            zone_id=selected_zone.entity_id, temps={self.device.entity_id: temperature}
+        )
+        _LOGGER.debug(
+            "Setting temperature for room %s to %s for zone %s",
+            self.device.entity_id,
+            temperature,
+            selected_zone.entity_id,
+        )
+
+        # manually set temperature in home assistant and netatmo
+        self._attr_target_temperature = temperature
+        for room in selected_zone.rooms:
+            if room.entity_id == self.device.entity_id:
+                room.therm_setpoint_temperature = temperature
```

Comment:
I don't know much about the pyatmo lib, but this looks like an attribute that should be updated internally by the lib (when it gets new data from the thermostat).

Reply:
I can understand that this is not a very nice of updating the state. However, if I don't do it immediately and let home assistant do the magic it takes too much time. Subsequent calls might operate on invalid state.

Reply:
The state will be updated asynchronously in the background. The manual override allows for multiple calls within a short span.